File: SereneMovie.Web/Modules/Default/Movie/MovieColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace SereneMovie.Default.Columns;

[ColumnsScript("Default.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }
    [DisplayName("Runtime in Mins")]
    public int Runtime { get; set; }
}

File: SereneMovie.Web/Modules/ServerTypes/Default/MovieColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieRow } from "./MovieRow";

export interface MovieColumns {
    Id: Column<MovieRow>;
    Title: Column<MovieRow>;
    Description: Column<MovieRow>;
    Storyline: Column<MovieRow>;
    Year: Column<MovieRow>;
    ReleaseDate: Column<MovieRow>;
    Runtime: Column<MovieRow>;
}

export class MovieColumns extends ColumnsBase<MovieRow> {
    static readonly columnsKey = 'Default.Movie';
    static readonly Fields = fieldsProxy<MovieColumns>();
}